packages/spectral/src/generators/componentManifest/constants.ts

```diff
@@ -1,36 +1,54 @@
 import path from "path";
 import { readJsonSync } from "fs-extra";
 
-export const PROCESS_ARGS = process.argv.slice(3);
-export const FLAG_DRY_RUN = PROCESS_ARGS.includes("--dry-run") ?? false;
-export const FLAG_INCLUDE_SIGNATURE = PROCESS_ARGS.includes(
-  "--include-signature"
-);
-export const FLAG_HELP = PROCESS_ARGS.includes("--help");
-export const FLAG_HELP_TEXT = [
-  { flag: "--help", description: "Show this help message." },
-  {
-    flag: "--dry-run",
+import { getFlagsStringValue } from "../utils/getFlagStringValue";
+import { getFlagsBooleanValue } from "../utils/getFlagBooleanValue";
+
+export const FLAGS = {
+  NAME: {
+    flag: ["--name", "-n"],
+    value: getFlagsStringValue(["--name", "-n"]),
+    description:
+      "The name of the component manifest. Defaults to the name of the current component being generated.",
+  },
+  OUTPUT_DIR: {
+    flag: ["--output-dir", "-o"],
+    value: getFlagsStringValue(["--output-dir", "-o"]),
+    description:
+      "The output directory for the component manifest. Defaults to the sibling of the current component directory.",
+  },
+  DRY_RUN: {
+    flag: ["--dry-run", "-d"],
+    value: getFlagsBooleanValue(["--dry-run", "-d"]),
     description:
       "Perform a dry run without generating the component manifest. This provides a preview of what you could expect to happen when running the command without this flag.",
   },
-  {
-    flag: "--include-signature",
+  INCLUDE_SIGNATURE: {
+    flag: ["--include-signature", "-s"],
+    value: getFlagsBooleanValue(["--include-signature", "-s"]),
     description:
       "This will include the published component's signature key. Allowing you to publish the Code Native Integrations (CNI) where this component manifest is used.",
   },
-];
+  HELP: {
+    flag: ["--help", "-h"],
+    value: getFlagsBooleanValue(["--help", "-h"]),
+    description: "Show this help message.",
+  },
+};
 
 export const COMPONENT_DIR = process.cwd();
+
 export const COMPONENT_DIST_DIR = path.join(COMPONENT_DIR, "dist", "index.js");
 
 export const SOURCE_DIR = path.join(__dirname, "templates");
-export const DESTINATION_BASE_NAME = `${path.basename(COMPONENT_DIR)}-manifest`;
-export const DESTINATION_DIR = path.join(
-  COMPONENT_DIR,
-  "..",
-  DESTINATION_BASE_NAME
-);
+
+export const DESTINATION_BASE_NAME = FLAGS.NAME.value
+  ? `${FLAGS.NAME.value}-manifest`
+  : `${path.basename(COMPONENT_DIR)}-manifest`;
```

Comment:
We should only append `-manifest` if there isn't a provided name. Otherwise, we should just use the name.